src/main/java/org/jenkinsci/test/acceptance/docker/fixtures/GitLabContainer.java

```diff
@@ -0,0 +1,115 @@
+package org.jenkinsci.test.acceptance.docker.fixtures;
+
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApiException;
+import org.gitlab4j.api.ProjectApi;
+import org.gitlab4j.api.models.Project;
+import org.jenkinsci.test.acceptance.docker.Docker;
+import org.jenkinsci.test.acceptance.docker.DockerContainer;
+import org.jenkinsci.test.acceptance.docker.DockerFixture;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.net.URI;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import org.jenkinsci.test.acceptance.po.CapybaraPortingLayer;
+import org.jenkinsci.test.acceptance.utils.ElasticTime;
+
+import static org.junit.Assert.assertTrue;
+
+
+@DockerFixture(id = "gitlab-plugin", ports = 22)
+public class GitLabContainer extends DockerContainer {
+    protected static final String REPO_DIR = "/home/gitlab/gitlabRepo";
+
+    private static final HttpClient client = HttpClient.newBuilder()
+                                                       .followRedirects(HttpClient.Redirect.NORMAL)
+                                                       .connectTimeout(Duration.ofSeconds(5))
+                                                       .build();
+
+    private static final ElasticTime time = new ElasticTime();
+
+    public String host() {
+        return ipBound(22);
+    }
+
+    public int port() {
+        return port(22);
+    }
+
+    public URL getURL() throws IOException {
+        // return new URL("http://" + host() + ":" + port());
+        return new URL("http://" + getIpAddress());
+    }
+
+    /** URL visible from the host. */
+    public String getRepoUrl() {
+        return "ssh://git@" + host() + ":" + port() + REPO_DIR;
+    }
+
+    @Deprecated
+    public String getRepoUrlInsideDocker() throws IOException {
+        return "ssh://git@" + getIpAddress() + REPO_DIR;
+    }
+
+    /**
+     * URL visible from other Docker containers.
+     * @param alias an alias for this container’s {@link #getCid} passed to {@code --link}
+     */
+    public String getRepoUrlInsideDocker(String alias) throws IOException {
+        return "ssh://git@" + alias + REPO_DIR;
+    }
+
+    public HttpResponse<String> createRepo(String repoName, String token) throws IOException {
+        try{
+            HttpRequest request = HttpRequest.newBuilder()
+                                             .uri(new URI("http://" + getIpAddress() + "/api/v4/projects"))
```

Comment:
I think this will not work in all scenarious - the web port has not been configured as exposed (`@DockerFixture(id = "gitlab-plugin", ports = 22)`) and as such you may not be able to communicate witht the web interface of GitLab depending on your docker setup.  IIUC you should add a new mapping for the web port.

Additionally you have `getURL()` that you should call to then addon the endpoint rather than repeating yourself.